### PR TITLE
Fetch & alert & modify obfuscated owners

### PR DESCRIPTION
**Pull request by annisassi**
# Hitas Pull Request

# Description

Bundles the data fetch of multiple obfuscated owners into one query. Deobfuscates apartment details data. Adds *** at the beginning of the obfuscated owner names. Replaces the name of obfuscated owners with *** in the apartment view. Adds tooltip to the obfuscate checkbox. Adds alert icon & text to the apartment details page when there are obfuscated owners on the apartment.

## Pull request checklist

Check the boxes for each DoD item that has been completed:

## Test plan

Check that:
- you see only *** for obfuscated owners in the apartments list view
- you see *** before the name of the obfuscated owners at codes: owners list, apartment details view and when doing a new trade or searching new owners for the apartment
- you can modify the data of an obfuscated owner
- you can access the tooltip for the obfuscate check box when modifying owner data
- you see an alert text & icon on the apartment view if there are obfuscated owners in the apartment

## Tickets

This pull request resolves all or part of the following ticket(s): HT-659

### CODE REVIEW

File: backend/poetry.lock
Comment (jforsman) [inline backend/poetry.lock:1]: You could try to check why category values have been removed by poetry when version has been changed from 1.4.2 to 1.6.1. Do not know if those have any influence on any deployments or not.

Comment (annisassi) [reply backend/poetry.lock:1]: Tested that the container starts up normally with Docker without categories, that are deprecated on poetry version 1.6.0-> The poetry version could be upgraded anyway, but it is another issue.

**Review by jforsman — APPROVED** [review]

File: frontend/src/features/apartment/ApartmentListPage.tsx [hunk 1]
```diff
@@ -18,7 +18,9 @@ import ConditionsOfSaleStatus from "./components/ConditionsOfSaleStatus";
 const ApartmentListItem = ({apartment}: {apartment: IApartment}): React.JSX.Element => {
     // Combine ownerships into a single formatted string
     const ownershipsString = apartment.ownerships.length
-        ? apartment.ownerships.map((o) => `${o.owner.name} (${o.owner.identifier})`).join(", ")
+        ? apartment.ownerships
+              .map((o) => (o.owner.non_disclosure ? "***" : `${o.owner.name} (${o.owner.identifier})`))
```
[the remark this answers is not in the document]
Comment (annisassi) [reply frontend/src/features/apartment/ApartmentListPage.tsx:22]: This is not the same function than formatOwner :) Here the name is replaced with *** while the formatOwner adds *** at the beginning of the name for obfuscated users.

File: frontend/src/common/services/hitasApi/apartment.ts [hunk 2]
```diff
@@ -37,6 +40,37 @@ const apartmentApi = hitasApi.injectEndpoints({
...
+                {type: "ObfuscatedOwner", id: arg},
+                {type: "Owner", id: arg},
+            ],
+        }),
+        getObfuscatedOwners: builder.query<{data: IOwner[]; error: FetchBaseQueryError[]}, string[]>({
+            queryFn: async (ids, _queryApi, _extraOptions, baseQuery) => {
+                const results: QueryReturnValue<unknown, FetchBaseQueryError>[] = await Promise.all(
+                    ids.map((id) => baseQuery(`owners/deobfuscated/${id}`))
+                );
+                const aEmptyData = [] as IOwner[];
```
[the remark this answers is not in the document]
Comment (annisassi) [reply frontend/src/common/services/hitasApi/apartment.ts:57]: Describing array variables... changed to more consistent variable names.

File: frontend/src/common/components/OwnershipList.tsx [hunk 3]
```diff
@@ -34,38 +34,40 @@ const OwnershipList = () => {
...
-                            <div className="icon--remove">
-                                <IconCrossCircle
-                                    size="m"
-                                    onClick={() => remove(index)}
-                                />
-                            </div>
-                        </li>
-                    ))}
+                    {fields.map((field, index) => {
+                        return (
```
[the remark this answers is not in the document]
Comment (annisassi) [reply frontend/src/common/components/OwnershipList.tsx:38]: This was left unintentionally after some testing... It's fixed now.

**Pull request merged**